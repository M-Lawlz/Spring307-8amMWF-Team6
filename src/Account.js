import App from "firebase/app";
import "firebase/auth";
import "firebase/firestore";
import React from "react";

export default class extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      user: null,
      userData: null,
    };
  }

  fetchUserInfo = () => {
    const userDoc = App.firestore()
      .collection("users")
      .doc(this.state.user.email);
    userDoc
      .get()
      .then((doc) => {
        if (doc.exists) {
          this.setState({ userData: doc.data() });
        }
      })
      .catch((error) => {
        console.error("Error: ", error.message);
      });
  };

  async componentDidMount() {
    await App.auth().onAuthStateChanged((user) => {
      if (user) {
        this.setState({ user: user });
        this.fetchUserInfo();
      }
    });
  }

  render() {
    if (!this.state.userData) {
      return null;
    } else {
      return (
        <div>
          <h1>Welcome, {this.state.userData.username}!</h1>
        </div>
      );
    }
  }
}
