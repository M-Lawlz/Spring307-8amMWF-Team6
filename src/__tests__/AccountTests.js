import Account from "../Account";
import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

test("Account renders successfully", () => {
  const account = shallow(<Account />);
  expect(toJson(account)).toMatchSnapshot();
});
