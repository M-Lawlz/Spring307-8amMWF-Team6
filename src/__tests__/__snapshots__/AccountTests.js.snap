// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account renders successfully 1`] = `""`;
